src/analytics/redux/tracked_events.js

```diff
@@ -64,20 +71,61 @@ function trackBlocksHandler(
 	} );
 }
 
+/**
+ * Helper function to track if a block is being dragged.
+ *
+ * @param {boolean} isDragging A boolean value to track if a block is being dragged
+ * @return {void}
+ */
+function setIsDraggingBlock( isDragging ) {
+	isDraggingBlock = isDragging;
+}
+
 /**
  * Helper function to track block movement events.
  *
- * @param {string} action Type of movement (arrows/drag & drop)
+ * @param {string[]} clientIds The client IDs of the blocks being moved
+ * @param {string}   action    Type of movement (arrows/drag & drop)
  * @return {void}
  */
-function trackBlockMoved( action ) {
+function trackBlockMoved( clientIds, action ) {
+	const block = select( 'core/block-editor' ).getBlocksByClientId(
+		clientIds?.[ 0 ]
+	)?.[ 0 ];
+
 	const eventProperties = {
 		action_source: action,
+		inner_block: !! block?.innerBlocks.length,
+		block_name: block?.name,
 	};
 
 	sendEventToHost( 'editor_block_moved', eventProperties );
 }
 
+/**
+ * Helper function to handle when a block is moved by an index position
+ *
+ * @param {string[]} clientIds The client IDs of the blocks being moved
+ * @param {number}   toIndex   Index of the new position of the block
+ * @return {void}
+ */
+function handleBlockMovedByPosition( clientIds, toIndex ) {
+	const numberOfBlocks = select( 'core/block-editor' ).getBlockCount() - 1;
```

Comment:
Nitpick: I understand that this constant holds the last block index instead of the number of blocks, so I'm wondering if we should rename it to something like `lastBlockIndex`, wdyt?

Reply:
Yep, I meant that the name `numberOfBlocks` might not be accurate as it doesn't hold the number of blocks because it's subtracting 1 to the result of `getBlockCount`.